File: Node/FirestoreTankDrive/tank/rosebot.js
const rosebotDriveSystem = require("./rosebotDriveSystem");
const rosebotServo = require("./rosebotServos");

// const test = new rosebotDriveSystem.DriveSystem();

class RoseBot {
    constructor() {
        // Motors
        this.driveSystem = new rosebotDriveSystem.DriveSystem();

        // Servos
        this.initializeServos();
    }

    async initializeServos() {
      const pca9685Driver = await rosebotServo.createPca9685Driver();
      this.armServos = new rosebotServo.ArmServos(pca9685Driver);
      this.gripperServo = new rosebotServo.GripperServo(pca9685Driver);
      this.cameraServo = new rosebotServo.CameraServo(pca9685Driver);
    }
}

function msleep(n) {
  Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, n);
}
function sleep(n) { 
  msleep(n*1000);
}

module.exports = {
    RoseBot,
    sleep,
    msleep
}

File: Python/RunsOnMyPi/rosebot.py
import rosebot_drive_system
import rosebot_servos
import rosebot_sensors

class RoseBot:
    def __init__(self):
        self.drive_system = rosebot_drive_system.DriveSystem()
        self.servos = rosebot_servos.RosebotServos()
        self.ultrasonic = rosebot_sensors.UltrasonicSensor()
        self.line = rosebot_sensors.LineSensors()